crates/hyle-modules/src/modules/prover.rs

```diff
@@ -104,65 +106,93 @@ pub enum AutoProverEvent<Contract> {
     /// Event sent when a blob is executed as success
     /// proof will be generated & sent to the node
     SuccessTx(TxHash, Contract),
+    /// Event sent in order to add an AutoProver for that contract
+    ProveContract(ContractName, Contract),
```

Comment:
I would use an other enum for this kind, `AutoProverEvent` are emited by the module, where `ProveContract` is sent to the module. For clarity I would have a dedicated enum `AutoProverCommand` for example